pkg/receive/hashring.go

```diff
@@ -235,6 +285,33 @@ func (m *multiHashring) GetN(tenant string, ts *prompb.TimeSeries, n uint64) (st
 	return "", errors.New("no matching hashring to handle tenant")
 }
 
+func hashringHasAZConfig(endpoints []string) bool {
+	for _, endpoint := range endpoints {
+		s := strings.Split(endpoint, ",")
+		if len(s) > 1 {
+			return true
+		}
+	}
+	return false
+}
+
+func validateAZHashring(endpoints []string, algorithm HashringAlgorithm) error {
+	hasAZConfig := hashringHasAZConfig(endpoints)
+	if !hasAZConfig {
+		return nil
+	}
+	if hasAZConfig && algorithm == AlgorithmHashmod {
+		return errors.New("Hashmod algorithm does not support AZ aware hashring configuration. Either use Ketama or remove AZ configuration.")
+	}
+	for _, endpoint := range endpoints {
+		s := strings.Split(endpoint, ",")
+		if len(s) == 1 {
+			return errors.New("Hashring contains a mix of AZ and non AZ aware endpoints.")
```

Comment:
Maybe we can check this in one go, in `hashringHasAZConfig` and return this error from there?

Reply:
Good point. I made some modifications so we don't have to iterate over the endpoints twice.